lib/msal-node/docs/key-vault-managed-identity.md

```diff
@@ -0,0 +1,222 @@
+
+```bash
+cat example.crt example.key > example.pem
+```
+
+> Powershell users can use **cat** equivalent below:
+>
+>```powershell
+>    Get-Content example.crt, exampleDecrypted.key | Set-Content example.pem
+>```
+
+This should give you `example.pem`. Next, **upload** this to Key Vault.
+
+1. Navigate to your key vault on [Azure portal](https://portal.azure.com).
+1. On the Key Vault properties pages, select **Certificates**.
+1. Click on **Generate/Import**.
+1. On the **Create a certificate** screen choose the following values:
+   * **Method of Certificate Creation**: Import.
+   * **Certificate Name**: ExampleCertificate.
+   * **Upload Certificate File**: select the certificate file from disk
+   * **Password** : If you are uploading a password protected certificate file, provide that password here. Otherwise, leave it blank. Once the certificate file is successfully imported, key vault will remove that password.
+1. Click **Create**.
+
+For alternative ways of importing, see: [Tutorial: Import a certificate in Azure Key Vault](https://docs.microsoft.com/azure/key-vault/certificates/tutorial-import-certificate).
+
+> :information_source: When you generate/import a certificate to Azure Key Vault Certificates, a corresponding private key is created automatically under Azure Key Vault Secrets. Later on, you can retrieve your private key from the **Secrets** blade.
+
+### Get certificate from your vault in Node.js
+
+Using [Azure Key Vault JavaScript SDK](https://docs.microsoft.com/javascript/api/overview/azure/keyvault-certificates-readme?view=azure-node-latest), we can fetch the certificate we've uploaded in the previous step. During development, Azure **Key Vault JavaScript SDK** grabs the required access token from the local environment using VS Code's context, via [@azure/identity](https://docs.microsoft.com/javascript/api/overview/azure/identity-readme?view=azure-node-latest) package. To do this, you'll need to be signed in to Azure.
+
+First, [download and install](https://docs.microsoft.com/cli/azure/install-azure-cli) **Azure CLI**. This should add **Azure CLI** to system path. Re-launch VS Code and **Azure CLI** should be available in VS Code [integrated terminal](https://code.visualstudio.com/docs/editor/integrated-terminal). Then, type the following to sign-in.
+
+```console
+az login --tenant YOUR_TENANT_ID
+```
+
+Once authenticated, [@azure/identity](https://docs.microsoft.com/javascript/api/overview/azure/identity-readme?view=azure-node-latest) package can access the Azure Key Vault as shown below:
+
+```JavaScript
+function msalApp(thumbprint, privateKey) {
+
+    // Before running the sample, you will need to replace the values in the config
+    const config = {
+        auth: {
+            clientId: "ENTER_CLIENT_ID",
+            authority: "https://login.microsoftonline.com/ENTER_TENANT_ID",
+            clientCertificate: {
+                thumbprint: thumbprint,
+                privateKey: privateKey,
```

Comment:
Hello team, sorry to chime in. Two questions:
- Is there a way to pass a non-PEM privateKey here?
- Is there a way to specify a password for the PEM certificate here?

Reply:
@sadasant sorry for late answer.
1) No, the privateKey should be PEM encoded
2) No again, but you can use the [crypto module](https://nodejs.org/docs/latest-v14.x/api/crypto.html#crypto_crypto_createprivatekey_key) to decrypt the password protected privateKey before passing it to msal-node:

```javascript
const fs = require('fs');
const crypto = require('crypto');
const privateKeySource = fs.readFileSync('./example.key')
const privateKeyObject = crypto.createPrivateKey({
    key: privateKeySource,
    passphrase: "YOUR_PASSPHRASE",
    format: 'pem'
});
const privateKey = privateKeyObject.export({
    format: 'pem',
    type: 'pkcs8'
});
```

cc @sameerag

Reply:
Thank you, @derisen